torch/export/_trace.py

```diff
@@ -887,280 +887,207 @@ def wrapper(*args, **kwargs):
-                        with gm.graph.inserting_before(node):
-                            new_node = gm.graph.create_node(
-                                "call_function",
-                                torch.ops.higher_order._export_tracepoint,
-                                args=node.args,
-                                kwargs={
-                                    "path": path,
-                                    "kind": node.kwargs["kind"],
-                                },
-                            )
-                            new_node.meta = node.meta
-                            node.replace_all_uses_with(new_node)
-                            gm.graph.erase_node(node)
-
-            res = CollectTracepointsPass(module_call_signatures, ep_non_strict.sig)(gm)
-            assert res is not None
-            gm = res.graph_module
-
-        _rewrite_non_persistent_buffers(mod, ep_non_strict.sig, ep_non_strict.constants)
-        _verify_nn_module_stack(gm)
-        _verify_stack_trace(gm)
-        _verify_placeholder_names(gm, ep_non_strict.sig)
-        exported_program = ExportedProgram(
-            root=gm,
-            graph=gm.graph,
-            graph_signature=ep_non_strict.sig,
-            state_dict=original_state_dict,
-            range_constraints=range_constraints,
-            module_call_graph=_make_module_call_graph(
-                _EXPORT_MODULE_HIERARCHY, orig_in_spec, out_spec, module_call_signatures
-            ),
-            example_inputs=(args, kwargs),
-            constants=ep_non_strict.constants,
-        )
-        return exported_program
+    gm.meta["forward_arg_names"] = forward_arg_names
+    module_call_signatures = {
+        strip_root(fqn): ModuleCallSignature(inputs=[], outputs=[], **specs)
+        for fqn, specs in module_call_specs.items()
+    }
+
+    if len(preserve_module_call_signature) > 0:
+        for node in gm.graph.nodes:
+            if node.target == torch.ops.higher_order._export_tracepoint:
+                if "path" in node.kwargs:
+                    path = strip_root(node.kwargs["path"])
+                    with gm.graph.inserting_before(node):
+                        new_node = gm.graph.create_node(
+                            "call_function",
+                            torch.ops.higher_order._export_tracepoint,
+                            args=node.args,
+                            kwargs={
+                                "path": path,
+                                "kind": node.kwargs["kind"],
+                            },
+                        )
+                        new_node.meta = node.meta
+                        node.replace_all_uses_with(new_node)
+                        gm.graph.erase_node(node)
+
+        res = CollectTracepointsPass(module_call_signatures, ep_non_strict.sig)(gm)
+        assert res is not None
+        gm = res.graph_module
+
+    _rewrite_non_persistent_buffers(mod, ep_non_strict.sig, ep_non_strict.constants)
```

Comment:
could we deduplicate this code too and put it in export_to_aten_ir? You don't have to necessarily do it in this PR, can do it in a followup.